shared/test/render-dumb-sheet.js

```diff
@@ -9,44 +9,62 @@ import DumbSheetItem from '../dev/dumb-sheet/item'
 
 const PADDING = 25
 
-ipcRenderer.on('display', (ev, msg) => {
-  const map = dumbComponentMap[msg.key]
-  const mockKey = msg.mockKey
+window.load = (options) => {
+  const onDisplay = (ev, msg) => {
+    const map = dumbComponentMap[msg.key]
+    const mockKey = msg.mockKey
 
-  const displayTree = (
-    <MuiThemeProvider muiTheme={materialTheme}>
-      <DumbSheetItem
-        key={mockKey}
-        id='rendered'
-        style={{alignSelf: 'flex-start', margin: PADDING}}
-        component={map.component}
-        mockKey={mockKey}
-        mock={map.mocks[mockKey]}
-      />
-    </MuiThemeProvider>
-  )
+    const displayTree = (
+      <MuiThemeProvider muiTheme={materialTheme}>
+        <DumbSheetItem
+          key={mockKey}
+          id='rendered'
+          style={{alignSelf: 'flex-start', margin: PADDING}}
+          component={map.component}
+          mockKey={mockKey}
+          mock={map.mocks[mockKey]}
+        />
+      </MuiThemeProvider>
+    )
 
-  const appEl = document.getElementById('root')
-  ReactDOM.render(displayTree, appEl, () => {
-    // Remove pesky blinking cursors
-    if (document.activeElement.tagName === 'INPUT') {
-      document.activeElement.blur()
+    const sendDisplayDone = () => {
+      // Unfortunately some resources lazy load after they're rendered.  We need
+      // to give the renderer time to load.  After trying process.nextTick,
+      // requestAnimationFrame, etc., simply putting in a time delay worked best.
+      setTimeout(() => {
+        const renderedEl = document.getElementById('rendered')
+        if (!renderedEl) {
+          ipcRenderer.send('display-error', {...msg})
+          return
+        }
+        const box = renderedEl.getBoundingClientRect()
+        const rect = {
+          x: box.left - PADDING,
+          y: box.top - PADDING,
+          width: Math.floor(box.width) + 2 * PADDING,
+          height: Math.floor(box.height + 2 * PADDING),
+        }
+
+        ipcRenderer.send('display-done', {rect, ...msg})
+      }, 1000)
     }
 
-    // Unfortunately some resources lazy load after they're rendered.  We need
-    // to give the renderer time to load.  After trying process.nextTick,
-    // requestAnimationFrame, etc., simply putting in a time delay worked best.
-    setTimeout(() => {
-      const renderedEl = document.getElementById('rendered')
-      const box = renderedEl.getBoundingClientRect()
-      const rect = {
-        x: box.left - PADDING,
-        y: box.top - PADDING,
-        width: Math.floor(box.width) + 2 * PADDING,
-        height: Math.floor(box.height + 2 * PADDING),
-      }
+    const appEl = document.getElementById('root')
+    try {
```

Comment:
we wrap this in a try and render an actual error so we can see whats going on easier